File: factorial/Factorial.java
import java.math.BigInteger;

public class Factorial {
    private static final int MOD = 100001;

    public static void main(String[] args) {
        int number = 5000000; // Change this value to calculate the factorial of a different number
        System.out.println("Factorial of " + number + " mod " + MOD + " is " + factorial(number));
    }

    public static BigInteger factorial(int n) {
        BigInteger result = BigInteger.ONE;
        for (int i = 1; i <= n; i++) {
            result = result.multiply(BigInteger.valueOf(i)).mod(BigInteger.valueOf(MOD));
        }
        return result;
    }
}

File: factorial/factorial.py
MOD = 100001

def factorial(n):
    result = 1
    for i in range(1, n + 1):
        result = (result * i) % MOD
    return result

number = 50000000  # Change this value to calculate the factorial of a different number
print(f"Factorial of {number} mod {MOD} is {factorial(number)}")
